tools/doc/apilinks.js

```diff
@@ -24,9 +24,18 @@ const path = require('path');
 const child_process = require('child_process');
 
 // determine orign repo and most recent commit
-const repo = (child_process.execSync('git remote get-url origin').toString()
-  .match(/(\w+\/\w+)\.git\r?\n?$/) || ['', 'nodejs/node'])[1];
-let hash = child_process.execSync('git log -1 --pretty=%H').toString().trim();
+const repo = (
```

Comment:
I think you should get this from the environment with a default of `https://github.com/nodejs/node/blob/`, similar to https://github.com/nodejs/node/blob/36468ca92862f2e1bc086be7e5e12ec8e012982b/Makefile#L606-L609

For example on my PC I don't have an `origin` remote, I have `upstream` instead. On the CI use use the `ssh` protocol so: `remote.origin.url=git@github.com:nodejs/node.git`

Reply:
I'm fine with an environment variable, and I'm fine with a default to `nodejs/node` when all else fails, and I'm fine with looking at `upstream` if `origin` is not found... that being said, I continue to assert that enabling a developer to validate their own work before creating a pull request is an important use case.

It looks like `git rev-parse --abbrev-ref --symbolic-full-name @{u}` will show the correct remote associated with the current branch.

Reply:
> that being said, I continue to assert that enabling a developer to validate their own work before creating a pull request is an important use case.

Sure, but I'm willing to assume the most common case, then allow for the developer to manually override it. Case in point our form to start a CI job. It has defaults, and allows the submitter to override them. IMHO trying to deduce configuration on end user's machine is just a world of hurt.


> It looks like git `rev-parse --abbrev-ref --symbolic-full-name @{u}` will show the correct remote associated with the current branch.

The trick is that we need to read files in this non checked out ref. Maybe we can stream them out, and only those who are diff from the current HEAD.

Reply:
Please explain _world of hurt_?

I found an alternate set of github commands that does the job, perhaps more clearly:

```
LOCAL_BRANCH=`git name-rev --name-only HEAD`
TRACKING_REMOTE=`git config branch.$LOCAL_BRANCH.remote`
REMOTE_URL=`git config remote.$TRACKING_REMOTE.url`
```

This (plus fallback to nodejs/node when any of the above commands fail) should work for the common case (CI, release).  It should also work when a developer is working on branch in their own private fork.

Why the insistence on inconveniencing the developer?